[Docs] Rewrite build documentation (#510)
oneapi-src/oneMKL: docs/building_and_running_tests.rst

.. _building_and_running_tests:

Building and Running Tests
==========================

> The functional are tests are enabled by default, and can be enabled/disabled

Reviewer: ```suggestion
The functional tests are enabled by default, and can be enabled/disabled
```

Author: Done in https://github.com/oneapi-src/oneMKL/pull/510/commits/4f6af02e17e392695cebf7b84af6a7a76ebb5d6f.

with the CMake build parameter ``-DBUILD_FUNCTIONAL_TESTS=True/False``. Only

> tests relevant for the enabled backends and target domains are built.

Reviewer: ```suggestion
the tests relevant to the enabled backends and target domains will be built.
```

Author: Done in https://github.com/oneapi-src/oneMKL/pull/510/commits/4f6af02e17e392695cebf7b84af6a7a76ebb5d6.

> Building tests for some domains may require additional libraries for reference.

Reviewer: ```suggestion
Building tests for BLAS and LAPACK domains requires additional libraries for reference.
```

Author: Done in https://github.com/oneapi-src/oneMKL/pull/510/commits/4f6af02e17e392695cebf7b84af6a7a76ebb5d6f.
